mcp/server.py

```
@@ -0,0 +1,542 @@
+@mcp.tool()
+def migrate_db(database_dsn: str | None = None) -> str:
+    """Run Sippy PostgreSQL migrations (``go run ./cmd/sippy migrate``).
+
+    Uses ``database_dsn`` when provided; otherwise ``SIPPY_DATABASE_DSN`` or a localhost dev default.
+    Full output is written to ``sippy-dev-logs/migrate_db.log``.
+    """
+    dsn = database_dsn or _default_database_dsn()
+    _ensure_dev_log_dir()
+    log_path = DEV_LOG_DIR / "migrate_db.log"
+    try:
+        r = subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)